### ISSUE TITLE
BT-723, OPT-155, OPT-156-LotResult: CVD vehicle category, type and number

### ISSUE THREAD

**Issue by odscjen** [open-contracting/european-union-support#127]
These 3 fields occur in `LotResult` and provide further information on the category (from [vehicle category codelist](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicle-category)), type (from [vehicles codelist](https://github.com/OP-TED/eForms-SDK/blob/develop/codelists/vehicles.gc)) and number of vehicles that a CVD award (BT-735-LotResult) covers.

From https://github.com/open-contracting/european-union-support/issues/122 we've mapped BT-735 to `tender.lots.additionalClassifications`. BT-723 could also map to that array with `.scheme` = "vehicle-category". OPT-155 and OPT-156 could map to `awards.items.classification` and `awards.items.quantity` respectively. 

There's two potential issues with this.
1. By mapping BT-723 to `tender.lots` and its siblings children to `awards` the information is split up, although `awards.relatedLots` does provide the link
2. In mapping BT-723 to `tender.lots` we're putting information that only appears once an award has been made into `tender`.

Any thoughts @jpmckinney

**Comment by jpmckinney** [comment]
I think we should map them all within award. Can we not make BT-723 an additional award item classification?

**Comment by odscjen** [comment]
Looking at this more closely the model goes:
 - BT-735 the CVD contract type, e.g. other service contract
     - BT-723 the vehicle category, e.g. M3 = buses
         - OPT-156 and OPT-155 the number and subtype of that vehicle, e.g. 5 zero-emission buses, 3 clean buses
```
 <efac:StrategicProcurement>
            <efac:StrategicProcurementInformation>
                <efbc:ProcurementCategoryCode listName="cvd-contract-type">oth-serv-contract</efbc:ProcurementCategoryCode>
                <efac:ProcurementDetails>
                    <efbc:AssetCategoryCode listName="vehicle-category">m3</efbc:AssetCategoryCode>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-zero-emissions</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>8</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
```

There can be multiple BT-735 in a lot (or LotResult), and for LotResults within each of them there can be multiple BT-723 and within each of those there can be multiple OPT-155/OPT-156 pairs.

As OPT-156 and OPT-155 are a quantity and associated classification they need to map to `awards.items.quantity` and `awards.items.classification` respectively.  So are you suggesting that BT-723 gets mapped to `awards.items.additionalClassification` for each child OPT-155?

The problem here is that there can be multiple instances of each code from OPT-155 under different BT-723's and the quantity for each could be different. So the above mapping would violate the guidance that items should not be duplicated as they would need to be. Tthey would have a different `additionalClassification` which would provide differentiation so maybe the violation in this case is acceptable.

None of these terms are mandatory, all are optional so there could be instances were BT-723 is given but nothing for OPT-155 and OPT156. Mapping to `additionalClassification` in that case would be odd...

**Comment by jpmckinney** [comment]
As I understand we can have:

```xml
<efac:StrategicProcurement>
    <efac:StrategicProcurementInformation>
        <efbc:ProcurementCategoryCode listName="cvd-contract-type">oth-serv-contract</efbc:ProcurementCategoryCode>
        <efac:ProcurementDetails>
            <efbc:AssetCategoryCode listName="vehicle-category">m3</efbc:AssetCategoryCode>
            <efac:StrategicProcurementStatistics>
                <efbc:StatisticsCode listName="vehicles">vehicles-zero-emissions</efbc:StatisticsCode>
                <efbc:StatisticsNumeric>8</efbc:StatisticsNumeric>
            </efac:StrategicProcurementStatistics>
            <efac:StrategicProcurementStatistics>
                <efbc:StatisticsCode listName="vehicles">vehicles-clean</efbc:StatisticsCode>
                <efbc:StatisticsNumeric>1</efbc:StatisticsNumeric>
            </efac:StrategicProcurementStatistics>
        </efac:ProcurementDetails>
        <efac:ProcurementDetails>
            <efbc:AssetCategoryCode listName="vehicle-category">m1</efbc:AssetCategoryCode>
            <efac:StrategicProcurementStatistics>
                <efbc:StatisticsCode listName="vehicles">vehicles-zero-emissions</efbc:StatisticsCode>
                <efbc:StatisticsNumeric>2</efbc:StatisticsNumeric>
            </efac:StrategicProcurementStatistics>
            <efac:StrategicProcurementStatistics>
                <efbc:StatisticsCode listName="vehicles">vehicles-clean</efbc:StatisticsCode>
                <efbc:StatisticsNumeric>3</efbc:StatisticsNumeric>
            </efac:StrategicProcurementStatistics>
        </efac:ProcurementDetails>
    </efac:StrategicProcurementInformation>
    <efac:StrategicProcurementInformation>
        <efac:ProcurementDetails>
            <efac:StrategicProcurementStatistics>
                <efbc:StatisticsNumeric>4</efbc:StatisticsNumeric>
            </efac:StrategicProcurementStatistics>
        </efac:ProcurementDetails>
    </efac:StrategicProcurementInformation>
</efac:StrategicProcurement>
```

The last `StatisticsNumeric` seems to be allowed by the eForms SDK, even if a quantity on its own is quite meaningless (just like a classification on its own is somewhat meaningless). This is just to say that as long as the OCDS mapping is not *more* meaningless, then we're okay :)

Now, originally I was thinking we might map this as:

```json
{
    "awards": [
        {
            "items": [
                {
                    "quantity": 8,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "m3"},
                        {"scheme": "vehicles", "id": "vehicles-zero-emissions"}
                    ]
                },
                {
                    "quantity": 1,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "m3"},
                        {"scheme": "vehicles", "id": "vehicles-clean"}
                    ]
                },
                {
                    "quantity": 2,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "m1"},
                        {"scheme": "vehicles", "id": "vehicles-zero-emissions"}
                    ]
                },
                {
                    "quantity": 3,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "m1"},
                        {"scheme": "vehicles", "id": "vehicles-clean"}
                    ]
                },
                {
                    "quantity": 4
                }
            ]
        }
    ]
}
```

As I understand, a vehicle is only one of m1, m2, etc. so it can't be counted twice at that level. However, that codelist also includes combinations like 'm1-m2-n1', so I suppose it's possible to double-count that way.

The [vehicles](https://github.com/OP-TED/eForms-SDK/blob/main/codelists/vehicles.gc) codelist has 'vehicles', 'vehicles-clean' and 'vehicles-zero-emission'. My understanding is that a zero-emission vehicle is a clean vehicle, but I don't know if the EU practice is to count separately or not.

The [cvd-contract-type](https://github.com/OP-TED/eForms-SDK/blob/main/codelists/cvd-contract-type.gc) codelist has codes with labels Other service contract, Passenger road transport services, and Vehicle purchase, lease or rent, which look to be non-overlapping.

We don't want double-counting in OCDS if we can avoid it. So, we could create a bespoke model just for the Clean Vehicle Directive. The other option is to accept that the codes 'm1-m2-n1' (Light-duty Vehicle) and 'n2-n3' (Truck) – if used with the other codes – could lead to double-counting; this option is more acceptable if the combination codes aren't expected to be used with the individual codes. The only remaining question for this second option is whether zero-emission vehicles are counted within vehicles-clean, or whether they are separated out.

**Comment by odscjen** [comment]
Ah, yes that modelling looks good.

The example in the [sdk pages](https://docs.ted.europa.eu/eforms/latest/schema/all-in-one.html#_lot_result) has:
```xml
<efac:NoticeResult>
    <!-- Interrupted Mark-up -->
    <efac:LotResult>
        <efac:StrategicProcurement>
            <efac:StrategicProcurementInformation>
                <efbc:ProcurementCategoryCode listName="cvd-contract-type">oth-serv-contract</efbc:ProcurementCategoryCode>
                <efac:ProcurementDetails>
                    <efbc:AssetCategoryCode listName="vehicle-category">n2-n3</efbc:AssetCategoryCode>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>8</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-zero-emission</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>0</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-clean</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>5</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                </efac:ProcurementDetails>
                <efac:ProcurementDetails>
                    <efbc:AssetCategoryCode listName="vehicle-category">m1</efbc:AssetCategoryCode>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>12</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-zero-emission</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>8</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-clean</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>4</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                </efac:ProcurementDetails>
            </efac:StrategicProcurementInformation>
            <efac:StrategicProcurementInformation>
                <efbc:ProcurementCategoryCode listName="cvd-contract-type">veh-acq</efbc:ProcurementCategoryCode>
                <efac:ProcurementDetails>
                <efbc:AssetCategoryCode listName="vehicle-category">m2</efbc:AssetCategoryCode>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>24</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-zero-emission</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>18</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-clean</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>6</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                </efac:ProcurementDetails>
                <efac:ProcurementDetails>
                    <efbc:AssetCategoryCode listName="vehicle-category">n1</efbc:AssetCategoryCode>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>6</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-zero-emission</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>6</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                    <efac:StrategicProcurementStatistics>
                        <efbc:StatisticsCode listName="vehicles">vehicles-clean</efbc:StatisticsCode>
                        <efbc:StatisticsNumeric>0</efbc:StatisticsNumeric>
                    </efac:StrategicProcurementStatistics>
                </efac:ProcurementDetails>
            </efac:StrategicProcurementInformation>
        </efac:StrategicProcurement>
        <!-- Interrupted Markup -->
    </efac:LotResult>
    <!-- Interrupted Mark-up -->
</efac:NoticeResult>
```

My reading is that this procurement is covering, 
* `another service contract` for 
  * 8 `n2-n3` vehicles, 0 of which are zero-emission and 5 of which are clean (and 3 presumably are neither)
  * 12 `m1` vehicles, 8 of which are zero-emission and 4 of which are clean
* `Vehicle purchase, lease or rent` for
  * 24 `m2` vehicles, 18 of which are zero-emissions and 6 of which are clean
  * 6 `n1` vehicles, 6 of which are zero-emissions and 0 of which are clean

Assuming they've used coherent data then: 
a) this looks like zero-emission and clean vehicles are different classifications and not overlapping ones and so we shouldn't get double counting from them.
b) to ensure the full quantity of vehicles is recorded we'll need to also include an item for vehicles that aren't zero-emission or clean when those values don't add up to the value given for `vehicles`.

So the OCDS using your modelling for the `another service contract` in the above example would look like:

```json
{
    "awards": [
        {
            "items": [
                {
                    "quantity": 3,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "n2-n3"}
                    ]
                },
                {
                    "quantity": 5,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "n2-n3"},
                        {"scheme": "vehicles", "id": "vehicles-clean"}
                    ]
                },
                {
                    "quantity": 8,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "m1"},
                        {"scheme": "vehicles", "id": "vehicles-zero-emissions"}
                    ]
                },
                {
                    "quantity": 4,
                    "additionalClassifications": [
                        {"scheme": "cvd-contract-type", "id": "oth-serv-contract"},
                        {"scheme": "vehicle-category", "id": "m1"},
                        {"scheme": "vehicles", "id": "vehicles-clean"}
                    ]
                }
            ]
        }
    ]
}
```
And there's no double counting.

My assumption for the combination codes, e.g. 'm1-m2-n1' is that in this case the procurement is for whichever of those types the supplier chooses to provide and that if there is another entry for just one of those codes that it would be a different part and so not double counting.

**Comment by jpmckinney** [comment]
Good catch on the full quantity of vehicles. Looks good!

**Comment by odscjen** [comment]
The guidance for this modelling is a bit tricky. I've had to incorporate BT-735-LotResult as well as OPT-155, OPT-156 and BT-723-LotResult. @duncandewhurst can you take a look and see if this is coherent?

> **OPT-155-LotResult**
> - If the value of `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsNumeric` is 0 discard. Otherwise:
> - [Get the award for the LotResult](common-operations.md#get-the-award-for-the-lot-result) and add an `Item` object to its `items` array.
       - Set the item's `.id` incrementally.
       - Add a `Classification` object to the item's `additionalClassifications` array.
       - Set the classification's `.scheme` to "vehicles".
       - Map the value of the field to the classification's `.id`.
       - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicles) and map it to the classification's `.description`.

> **OPT-156-LotResult**
> These values are mapped to the same `Item` objects as created for OPT-155.

> If the value of this field is 0 then discard. Otherwise:
> If the value of the `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsCode` is “vehicles-zero-emissions” or “vehicles-clean”, get the corresponding item and map to it’s `.quantity`. 
> If the value of `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsCode` is “vehicles”:
get the values of `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsNumeric` for `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsCode` equals “vehicles-zero-emissions” or “vehicles-clean”
> - subtract these values from the value of this field
> - map this new value to the item’s `.quantity`.


>**BT-723-LotResult**
> These values are mapped to the same `Item` objects as created for OPT-155-LotResult.

> For each `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsCode` add a `Classification` object to the corresponding item’s `additionalClassifications` array. 
 	- Set the classification's `.scheme` to "vehicle-category".
       	- Map the value of the field to the classification's `.id`.
       	- Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicle category) and map it to the classification's `.description`.

> If OPT-155-LotResult is absent for `ancestor::efac:ProcurementDetails` then [Get the award for the LotResult](common-operations.md#get-the-award-for-the-lot-result) and add an `Item` object to its `items` array.
       	- Set the item's `.id` incrementally.
       	- Add a `Classification` object to the item’s `additionalClassifications` array.
 	- Set the classification's `.scheme` to "vehicle-category".
       	- Map the value of the field to the classification's `.id`.
       	- Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicle category) and map it to the classification's `.description`.

>**BT-735-LotResult**
> These values are mapped to the same `Item` objects as created for OPT-155-LotResult and BT-723-LotResult.

> For each `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efac:StrategicProcurementStatistics` and `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efbc:AssetCategoryCode`
       - Add a `Classification` object to the item's `additionalClassifications` array.
       - Set the classification's `.scheme` to "cvd-contract-type".
       - Map the value of the field to the classification's `.id`.
       - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/cvd-contract-type) and map it to the classification's `.description`.

> If OPT-155-LotResult and BT-723-LotResult are absent for `ancestor::efac:ProcurementDetails` then [Get the award for the LotResult](common-operations.md#get-the-award-for-the-lot-result) and add an `Item` object to its `items` array.
       	- Set the item's `.id` incrementally.
       	- Add a `Classification` object to the item’s `additionalClassifications` array.
       - Set the classification's `.scheme` to "cvd-contract-type".
       - Map the value of the field to the classification's `.id`.
       - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/cvd-contract-type) and map it to the classification's `.description`.

**Comment by duncandewhurst** [comment]
**OPT-155-LotResult**

> If the value of `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsNumeric` is 0 discard. Otherwise:

I think this should be `ancestor::efac:StrategicProcurementStatistics/efbc:StatisticsNumeric`.

**OPT-156-LotResult**, **BT-723-LotResult**, **BT-735-LotResult**

> These values are mapped to the same...

Let's use active voice and refer to fields rather than values: "This field maps to the same...". We might need to make the same change for other mappings, too.

**OPT-156-LotResult**

> If the value of `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsCode` is “vehicles”: get the values of `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsNumeric` for `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsCode` equals “vehicles-zero-emissions” or “vehicles-clean”
> 
> * subtract these values from the value of this field
> 
> * map this new value to the item’s `.quantity`.

Suggest rewording as:

> If the value of `ancestor::efbc:StrategicProcurementStatistics/efbc:StatisticsCode` is “vehicles”, get the value of `/efbc:StatisticsNumeric` for each `ancestor::efbc:StrategicProcurementStatistics` with `/efbc:StatisticsCode` equal to “vehicles-zero-emissions” or “vehicles-clean” and subtract the sum of the values from the value of this field. Map the the result to the item's `.quantity`.

**BT-723-LotResult**

> - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicle category) and map it to the classification's `.description`.

Should be:

> - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicle-category) and map it to the classification's `.description`.

Unless I missed some nuance, I think the instructions can be refactored to reduce repetition:

> These values are mapped to the same `Item` objects as created for OPT-155-LotResult. If no `Item` objects were created for `ancestor::efac:ProcurementDetails`:
> * [Get the award for the LotResult](common-operations.md#get-the-award-for-the-lot-result)
> * Add an `Item` object to its `items` array.
> * Set the item's `.id` incrementally.
>
> For each `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsCode` add a `Classification` object to the corresponding item’s `additionalClassifications` array.
> - Set the classification's `.scheme` to "vehicle-category".
> - Map the value of the field to the classification's `.id`.
> - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/vehicle-category) and map it to the classification's `.description`.

**BT-735-LotResult**

Unless I misunderstood, I think `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efbc:AssetCategoryCode` should be `efac:StrategicProcurement/efac:StrategicProcurementInformation/efbc:ProcurementCategoryCode`.

I don't understand why `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efac:StrategicProcurementStatistics` is referenced in the mapping.

I think that the mapping can be refactored along similar lines to the change I proposed to 
BT-723-LotResult.

**Comment by odscjen** [comment]
thanks @duncandewhurst, all your suggestions for OPT-155, OPT-156 and BT-723 look good.

wrt BT-735-LotResult, `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efbc:AssetCategoryCode` is needed because that is where BT-723 is found and there needs to be a unique `item` for each combination of BT-723 and OPT-155. Otherwise I think you're right that it can be refactored in a similar way to how you've rewritten the BT-723 guidance.

> **BT-735-LotResult**
> This field maps to the same `Item` objects as created for OPT-155-LotResult and BT-723-LotResult. If no `Item` objects were created for `ancestor::efac:ProcurementDetails`:
> * [Get the award for the LotResult](common-operations.md#get-the-award-for-the-lot-result)
> * Add an `Item` object to its `items` array.
> * Set the item's `.id` incrementally.
>
> For each combination of `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsCode` and `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efbc:AssetCategoryCode` add a `Classification` object to the corresponding item’s `additionalClassifications` array.
> - Set the classification's `.scheme` to "cvd-contract-type".
> - Map to the classification's `.id`.
> - Look up the code's label in the [authority table](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/cvd-contract-type) and map it to the classification's `.description`.

Where I've incorporated your suggestion about active voice in the first sentence and the "Map to..." decision from https://github.com/open-contracting/european-union-support/issues/151#issuecomment-1360418976

**Comment by duncandewhurst** [comment]
> If no `Item` objects were created for `ancestor::efac:ProcurementDetails`:

I don't think this is valid for BT-735-LotResult since `efac:ProcurementDetails` does not appear in `xpathAbsolute`.

> For each combination of `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsCode` and `ancestor::efac:StrategicProcurementInformation/efac:ProcurementDetails/efbc:AssetCategoryCode` add a `Classification` object to the corresponding item’s `additionalClassifications` array.

The mapping for OPT-155-LotResult creates an `Item` for each `efac:StrategicProcurementStatistics` object so isn't it sufficient to say the following?

> For each `ancestor::efac:ProcurementDetails/efac:StrategicProcurementStatistics/efbc:StatisticsCode` add a `Classification` object to the corresponding item’s `additionalClassifications` array.

**Comment by odscjen** [comment]
> I don't think this is valid for BT-735-LotResult since efac:ProcurementDetails does not appear in xpathAbsolute.

you're right, I mistyped the xpath, it should be

> If no `Item` objects were created for `ancestor::efac:StrategicProcurementInformation/efbc:ProcurementDetails`

and

> The mapping for OPT-155-LotResult creates an Item for each efac:StrategicProcurementStatistics object so isn't it sufficient to say the following?

Yes looking at it again I think you're right.

the only other change I think is needed is to the end of the BT-156 guidance where if the calculated number is 0 then the item should be discarded completely.

**Comment by odscjen** [comment]
I've updated the guidance.yaml for these 4 BT's (BT-723-LotResult, BT-735-LotResult, OPT-155-LotResult and OPT-156-LotResult). I'm leaving this issue open as a reminder to write up a worked example as per #152

**Comment by duncandewhurst** [comment]
All sounds good. I'm sure the worked example will be helpful :-)

**Comment by duncandewhurst** [comment]
Closing in favour of https://github.com/open-contracting/european-union-support/issues/152#issuecomment-1409981079
